modules/services/window-managers/spectrwm.nix

```
@@ -0,0 +1,142 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.xsession.windowManager.spectrwm;
+
+  renderedValue = val:
+    if isBool val then if val then "1" else "0" else toString val;
+
+  renderedSettings = settings:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "${name} = ${renderedValue value}") settings);
+
+  renderedBindings = bindings:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "bind[${name}] = ${value}") bindings);
+
+  renderedUnbindings = unbindings:
+    concatStringsSep "\n" (map (value: "bind[] = ${value}") unbindings);
+
+  renderedPrograms = programs:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "program[${name}] = ${value}") programs);
+
+  renderedQuirks = quirks:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "quirk[${name}] = ${value}") quirks);
+
+  settingType = types.oneOf [ types.bool types.int types.str ];
+
+in {
+  meta.maintainers = [ hm.maintainers.loicreynier ];
+
+  options = {
+    xsession.windowManager.spectrwm = {
+      enable = mkEnableOption "Spectrwm window manager";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.spectrwm;
+        defaultText = literalExpression "pkgs.spectrwm";
+        description = ''
+          Package providing the <command>spectrwm</command> command.
+        '';
+      };
+
+      settings = mkOption {
+        type = types.attrsOf settingType;
+        default = { };
+        example = literalExpression ''
+          {
+            modkey = "Mod4";
+            workspace_limit = 5;
+            focus_mode = "manual";
+            focus_close = "next";
+          }
+        '';
+        description = "Spectrwm settings.";
+      };
+
+      bindings = mkOption {
+        type = types.attrsOf types.string;
```

Instead of `types.string` use `types.str` which has more predictable behavior:

```suggestion
        type = types.attrsOf types.str;
```